### ISSUE TITLE
TS2363: Error arithmetic operation while using Number object

### ISSUE THREAD

**Issue by vicmarcal** [microsoft/TypeScript#48762]
# Bug Report

Typescript is reporting an error when trying to perform an arithmetic operation between a "number" - "Number".

### 🔎 Search Terms

TS2363 , Number

### 🕗 Version & Regression Information

Happens in Playground

### ⏯ Playground Link

[Playground link with relevant code][(https://www.typescriptlang.org/play?#code/PTAEFkE9QYwewCYFNQHM5IM6gBZIE5JA)](https://www.typescriptlang.org/play?ts=4.7.0-dev.20220419#code/PTAEHUFMBsGMHsC2lQBd5oBYoCoE8AHSAZVgCcBLA1UABWgEM8BzM+AVwDsATAGiwoBnUENANQAd0gAjQRVSQAUCEmYKsTKGYUAbpGF4OY0BoadYKdJMoL+gzAzIoz3UNEiPOofEVKVqAHSKymAAmkYI7NCuqGqcANag8ABmIjQUXrFOKBJMggBcISGgoAC0oACCbvCwDKgU8JkY7p7ehCTkVDQS2E6gnPCxGcwmZqDSTgzxxWWVoASMFmgYkAAeRJTInN3ymj4d-jSCeNsMq-wuoPaOltigAKoASgAywhK7SbGQZIIz5VWCFzSeCrZagNYbChbHaxUDcCjJZLfSDbExIAgUdxkUBIursJzCFJtXydajBBCcQQ0MwAUVWDEQC0gADVHBQGNJ3KAALygABEAAkYNAMOB4GRonzgu4aJx2IhpN8AIw80AAVgA3IoZf1IBIAHLyxXY3mcPWgQ0K74ACgAzABKLWKHSOUAEqI0U1G5WgADUuoN3rI5Magng7gCIuY1tp9MZ7lZlA57ntwRUOGaHjIXkQ4ucwPYNC+bjMzHYDGYkH4sGg6kSHPgehEXj5dIZTMEfKS2L54AcqAA5MJ9Xq+UEVAB5L5kd6CKtaSBHVCOBSuaR4N2QXM6YZYfQoBCIaHCS7FiTi6IiAwcbELJisDg8IIlIA)

### 💻 Code

```ts
let number1 = 5;
let newNumber = new Number(3);
var result = number1 + newNumber
```

### 🙁 Actual behavior

It is preventing to use a sum with a Number

### 🙂 Expected behavior

It should probably let to perform a sum with the Number

**Comment by MartinJohns** [comment]
IMO TypeScript is protecting you from doing something silly like `new Number(3)`.

**Comment by RyanCavanaugh** [comment]
See also #2361
